tests/apps/weather/weather_test.py

```diff
@@ -0,0 +1,307 @@
+    return key
+
+
+@pytest.mark.asyncio
+async def test_get_geometric_info_by_address(google_api_key):
+    full_address, lat, lng = await get_geometric_info_by_address(
+        addr1,
+        google_api_key,
+    )
+
+    assert full_address == "대한민국 경기도 부천시"
+    assert lat == 37.5034138
+    assert lng == 126.7660309
+
+    full_address, lat, lng = await get_geometric_info_by_address(
+        addr2,
+        google_api_key,
+    )
+
+    assert full_address == "대한민국 서울특별시"
+    assert lat == 37.566535
+    assert lng == 126.9779692
+
+
+@pytest.mark.asyncio
+async def test_get_weather_wrong_geometric_info(response_mock):
+    addr = "WRONG"
+    key = "XXX"
+    response_mock.get(
+        "https://maps.googleapis.com/maps/api/geocode/json"
+        f"?region=kr&address={addr}&key=XXX",
+        body=json.dumps(
+            {
+                "results": [],
+            }
+        ),
+        headers={"Content-Type": "application/json"},
+    )
+    with pytest.raises(IndexError):
+        await get_geometric_info_by_address(
+            addr,
+            key,
+        )
+
+
+@pytest.mark.asyncio
+async def test_get_weather_with_wrong_openweather_coordination(response_mock):
+    response_mock.get(
+        "https://api.openweathermap.org/data/2.5/weather?"
+        "lat=123&lon=456&appid=asdf&units=metric",
+        body=json.dumps({}),
+        status=404,
+        headers={"Content-Type": "application/json"},
+    )
+
+    with pytest.raises(WeatherResponseError):
+        await get_weather_by_coordinate(123, 456, "asdf")
+
+
+@pytest.mark.asyncio
+async def test_get_air_pollution_with_wrong_coordination(response_mock):
+    response_mock.get(
+        "https://api.openweathermap.org/data/2.5/air_pollution?"
+        "lat=123&lon=456&appid=asdf",
+        body=json.dumps({}),
+        status=404,
+        headers={"Content-Type": "application/json"},
+    )
+
+    with pytest.raises(WeatherResponseError):
+        await get_air_pollution_by_coordinate(123, 456, "asdf")
+
+
+def test_get_aqi_description():
+    assert get_aqi_description(0).startswith("좋음")
+    assert get_aqi_description(1).startswith("좋음")
+    assert get_aqi_description(2).startswith("보통")
+    assert get_aqi_description(3).startswith("민감군 영향")
+    assert get_aqi_description(4).startswith("나쁨")
+    assert get_aqi_description(5).startswith("매우 나쁨")
+    # API Spec은 5단계가 최대입니다.
+    assert get_aqi_description(6).startswith("매우 나쁨")
+
+
+def test_degree_to_direction():
+    cases = [
+        (0, "N"),
+        (22.5, "NNE"),
+        (45, "NE"),
+        (67.5, "ENE"),
+        (90, "E"),
+        (112.5, "ESE"),
+        (135, "SE"),
+        (157.5, "SSE"),
+        (180, "S"),
+        (202.5, "SSW"),
+        (225, "SW"),
+        (247.5, "WSW"),
+        (270, "W"),
+        (292.5, "WNW"),
+        (315, "NW"),
+        (337.5, "NNW"),
+    ]
+
+    for degree, direction in cases:
+        assert direction == degree_to_direction(degree)
```

Comment:
이런 상황에선 https://docs.pytest.org/en/6.2.x/parametrize.html 를 사용하면 테스트가 분할되어서 오류 파악이 한결 용이해집니다.

```suggestion
@pytest.mark.parametrize(
    "degree,direction",
    [
        (0, "N"),
        (22.5, "NNE"),
        (45, "NE"),
        (67.5, "ENE"),
        (90, "E"),
        (112.5, "ESE"),
        (135, "SE"),
        (157.5, "SSE"),
        (180, "S"),
        (202.5, "SSW"),
        (225, "SW"),
        (247.5, "WSW"),
        (270, "W"),
        (292.5, "WNW"),
        (315, "NW"),
        (337.5, "NNW"),
    ],
)
def test_degree_to_direction(degree, direction):
    assert direction == degree_to_direction(degree)
```